src/server/status_server/mod.rs

```diff
@@ -408,47 +420,143 @@ impl StatusServer {
         )
     }
 
+    pub fn dump_region_meta(
+        req: Request<Body>,
+        router: Option<&RaftRouter<RocksEngine>>,
+    ) -> Box<dyn Future<Item = Response<Body>, Error = hyper::Error> + Send> {
+        lazy_static! {
+            static ref REGION: Regex = Regex::new(r"/region/(?P<id>\d+)").unwrap();
+        }
+
+        let err_resp =
+            |status_code, body| Box::new(ok(StatusServer::err_response(status_code, body)));
+
+        let not_found = || {
+            err_resp(
+                StatusCode::NOT_FOUND,
+                format!("{} not found", req.uri().path()),
+            )
+        };
+
+        let router = match router {
+            None => {
+                return err_resp(
+                    StatusCode::INTERNAL_SERVER_ERROR,
+                    "router not set".to_string(),
+                )
+            }
+            Some(r) => r,
+        };
+
+        match REGION.captures(req.uri().path()) {
+            None => not_found(),
+            Some(cap) => {
+                let id: u64 = match cap["id"].parse() {
+                    Ok(id) => id,
+                    Err(err) => {
+                        return err_resp(
+                            StatusCode::BAD_REQUEST,
+                            format!("invalid region id: {}", err),
+                        );
+                    }
+                };
+                let (tx, rx) = oneshot::channel();
+                match CasualRouter::send(
+                    router,
+                    id,
+                    CasualMessage::Test(Box::new(move |peer| {
+                        if let Err(meta) = tx.send(region_meta::RegionMeta::new(&peer.peer)) {
+                            error!("receiver dropped, region meta: {:?}", meta)
+                        }
+                    })),
+                ) {
+                    Ok(_) => (),
+                    Err(raftstore::Error::RegionNotFound(_)) => return not_found(),
+                    Err(err) => {
+                        return err_resp(
+                            StatusCode::INTERNAL_SERVER_ERROR,
+                            format!("channel pending or disconnect: {}", err),
+                        )
+                    }
```

Comment:
we should not return `StatusCode::NOT_FOUND` here, since we are already accepted and handling the request.

Reply:
I think `NOT_FOUND` is OK, but better with a more precise error description in response body.

Reply:
I'm OK with `StatusCode::NOT_FOUND` too, as long as there are more precise error description.

Reply:
More precise error description is added.